packages/tasks_panel/test/misc/fiori-e2e-config.spec.ts

```diff
@@ -380,56 +223,62 @@ specVersion: '2.4'
         .expects("createFileSystemWatcher")
         .withExactArgs(new testVscode.RelativePattern(wsFolder, "/project/ui5-deploy.yaml"), false, false, true)
         .returns(watcher);
+      mockE2eConfig = sandbox.mock(e2eConfig);
     });
 
     afterEach(() => {
-      tasks = [];
+      mockE2eConfig.verify();
     });
 
+    const data = { wsFolder, project };
     it("fioriE2eConfig, deploy.yaml is not updated - timeout reached", async () => {
-      expect(await fioriE2eConfig(wsFolder, project)).to.be.undefined;
+      expect(await fioriE2eConfig(data)).to.be.undefined;
     }).timeout(6000);
 
     it("fioriE2eConfig, deploy.yaml is updated, target ABAP, task defined", async () => {
-      tasks.push({ label: "test" });
       setTimeout(() => {
         callbacks["change"]();
       });
       mockWorkspaceFs
         .expects("readFile")
         .withExactArgs(uriUi5DeployYaml)
         .resolves(Buffer.from(ui5DeployYamlAbap, `utf8`));
-      mockWorkspace
-        .expects("getConfiguration")
-        .withExactArgs("tasks", testVscode.Uri.file(wsFolder))
-        .returns(taskConfig);
+      mockE2eConfig.expects("addTaskDefinition").resolves();
+      mockCommands.expects("executeCommand").withArgs("tasks-explorer.editTask").resolves();
       mockCommands.expects("executeCommand").withArgs("tasks-explorer.tree.select").resolves();
-      await fioriE2eConfig(wsFolder, project);
-      expect(/^Deploy to ABAP$/.test((tasks[1] as any).label)).to.be.true;
-      expect((tasks[1] as any).type).to.be.equal("npm");
-      expect((tasks[1] as any).options).to.be.deep.equal({
+      await fioriE2eConfig(data);
+      const tasks = (mockE2eConfig as any).expectations["addTaskDefinition"][0].args[0][1];
```

Comment:
if you really need `any` variable, at least do the cast once and assign to another variable
instead of ~10 times in the next 10 lines